packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/BaseViewManager.java

```diff
@@ -40,7 +42,32 @@
  * provides support for base view properties such as backgroundColor, opacity, etc.
  */
 public abstract class BaseViewManager<T extends View, C extends LayoutShadowNode>
-    extends ViewManager<T, C> implements BaseViewManagerInterface<T> {
+    extends ViewManager<T, C> implements BaseViewManagerInterface<T>, View.OnLayoutChangeListener {
+
+  String transformOrigin = null;
+  ReadableArray transformMatrix = null;
+
+  @Override
+  public void onLayoutChange(View v,
+                               int left,
+                               int top,
+                               int right,
+                               int bottom,
+                               int oldLeft,
+                               int oldTop,
+                               int oldRight,
+                               int oldBottom) {
+      // Old width and height
+      int oldWidth = oldRight - oldLeft;
+      int oldHeight = oldBottom - oldTop;
+
+      // Current width and height
+      int currentWidth = right - left;
+      int currentHeight = bottom - top;
+      if (currentHeight != oldHeight || currentWidth != oldWidth) {
+        applyTransformWithTransformOrigin((T) v, this.transformOrigin);
+      }
+  }
```

Comment:
Don't use a layout listener, use the hooks RN already uses for setting the layout calculated elsewhere.

Reply:
i tried finding a better way for android but looks like the layout gets set in SurfaceMountingManager [here](https://github.com/facebook/react-native/blob/2d07d5f160efdd25f5b3dbfa65c13884df9f3117/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/fabric/mounting/SurfaceMountingManager.java#L953) that doesn't invoke any hooks that I can listen in the view or manager. Let me know if I am missing something. Thanks.